### ISSUE TITLE
Deep copy kenlm.State, not shallow copy

### ISSUE THREAD

**Issue by igormq** [kpu/kenlm#228]
Unfortunately, we can't deepcopy the state

```python
import copy
s1 = kenlm.State()
s2 = kenlm.State()

s3 = copy.copy(s1)

m.BeginScore(s1, 'word', s2)

assert s1 == s2 (False)
assert s1 == s3 (True)

s3 = copy.deepcopy(s1)
>> TypeError: self._c_state cannot be converted to a Python object for pickling
```

**Comment by kpu** [comment]
There's no object recursion in State, so I guess you want `__deepcopy__` to do the same thing as `__copy__`?

**Comment by kpu** [comment]
Fixed in f39edc3
